oauth2client/service_account.py

```diff
@@ -461,3 +465,100 @@ def create_delegated(self, sub):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+
+
+class JWTAccessCredentials(GoogleCredentials):
+    """Self signed JWT credentials.
+
+    Makes an assertion to server using a self signed JWT from service account
+    credentials.  These credentials do NOT use OAuth 2.0 and instead authenticate
+    directly.
+    """
+    MAX_TOKEN_LIFETIME_SECS = 3600
+    """Max lifetime of the token (one hour, in seconds)."""
+
+    def __init__(self, service_account_creds, additional_claims=None, access_token=None,
+                 token_expiry=None):
+        super(JWTAccessCredentials, self).__init__(access_token, None, None, None, token_expiry,
+                                                   None, None)
+        self._creds = service_account_creds
+        self._claims = {}
+        self._claims.update(service_account_creds._kwargs)
+        if additional_claims:
+          self._claims.update(additional_claims)
+    
+    def get_access_token(self, http=None, additional_claims=None):
+        """Create a signed jwt
+        Agrs:
+            additional_claims: dict, additional claims to add 
+                the payload of the JWT. 
+            Returns:
+                An AccessTokenInfo with the signed jwt
+        """
+        if additional_claims is not None:
+          # Create a 1 time token
+          return self._create_token(additional_claims)
+        else:
+          if self.access_token is None or self.access_token_expired:
+            self._refresh(None)
+
+        return AccessTokenInfo(access_token=self.access_token,
+                              expires_in=self._expires_in())
+    def revoke(self, http):
+        # TODO(kpayson): Is there any way to actually invalidate a signed JWT?
```

Comment:
Please don't put todos in source files. And no, there is not way to invalidate a JWT. I think this should just pass, I don't think we need to raise an error here.